Drivers/Periphs/Src/radio.c

```diff
@@ -0,0 +1,59 @@
+#include "radio.h"
+
+#define CMD_SIZE 64
+
+struct sockaddr_in radioSocket;
+int lSocket;
+
+void Radio_Init(void){
+  MX_LWIP_Init();
+  //initialize socket to TCP protocol with byte streams
+  lSocket = lwip_socket(AF_INET, SOCK_STREAM, 0);
+  if (lSocket < 0) {/*do something here*/}
+  //Initialize Radio Socket address to TCP with IP address 0.0.0.0, port 23
+  radioSocket.sin_family = AF_INET;
+  radioSocket.sin_len = sizeof(radioSocket);
+  radioSocket.sin_addr.s_addr = htonl(INADDR_ANY);
+  radioSocket.sin_port = htons(23);
+  //bind socket to local address
+  if(lwip_bind(lSocket, (struct sockaddr*)&radioSocket, sizeof(radioSocket)) < 0) {
+    lwip_close(lSocket);
+    return;
+  }
+  //marks socket to accept connection requests (by radio)
+  if(lwip_listen(lSocket, 20) != 0) {
+    lwip_close(lSocket);
+    return;
+  }
+}
+
+ErrorStatus Radio_RX(void* data){
+  int addr, nbytes;
+  char buffer[1024];
+  struct sockaddr_in client_addr;
+  socklen_t size = sizeof(client_addr);
+  int error = lwip_connect(lSocket, (struct sockaddr*)&radioSocket, sizeof(radioSocket)); //make connection
+  if (error < 0){return ERROR;/*DO SOMETHING HERE*/} //This means it couldn't connect
+  addr = lwip_accept(lSocket, (struct sockaddr*)&client_addr, &size);
```

Comment:
If we're using TCP sockets, then we probably shouldn't be `accept()`ing a new connection and then `close()`ing that connection every time we want to send some information. The usual thing to do would be to accept a connection initially, and then send and receive to/from that socket until all communication ceases. There's a few ways to do this that might make sense:

1. Get a single client, and start providing data to them. If the client wants to send information back, then receive information from them at various points. Eventually, the client (the computer receiving the info) will probably be the one to terminate the connection.
2. Another option is to spawn a thread which will do the following: `accept()` a client, spawn a new thread to deal with sending/receiving to/from that client, and then loop to see if another client is connected. This has the benefit of being able to support multiple clients, but requires thread management, which might be difficult to manage with freertos, especially since (I think) we don't have an allocator.
3. One somewhat appealing option might be to use UDP instead of TCP. This has the downside of not including the error correction measures that TCP includes, but UDP allows us to simply 'broadcast' messages, which means the telemetry system in the car wouldn't have to track connections to clients, and clients would just listen for the broadcast when it was sent. If we did this, it probably wouldn't be feasible to carry out back-and-forth communication between Sunlight and the client computer that would be listening to all this.

Interested in seeing what others think about this @manthanand @ClarkPoon